python/orca/src/bigdl/orca/learn/pytorch/callbacks/maincallback.py

```diff
@@ -93,6 +93,7 @@ def on_val_forward(self, runner):
         Any behavior inconsistent with the default training behavior should be overridden here.
         """
         self.on_iter_forward(runner)
+        runner.target = runner.batch[-1]
```

Comment:
```python
class MainCallback(Callback):
    """
    MainCallback is a one-of-a-kind callback that contains hook functions:
        - `on_iter_forward`
        - `on_iter_backward`
        - `on_lr_adjust`

    These methods are somewhat special, because only one special MainCallback
    should be allowed to implement these methods among all callbacks, otherwise
    there will propagate forward and backward twice.
    """
    def on_iter_forward(self, runner):
        """
        If `on_train_forward` and `on_val_forward` are not overridden,
        this will be called during forward when training and validating.
        Any behavior inconsistent with the default forward behavior should be overridden here.
        """
        # unpack features into features and targets
        *features, target = runner.batch
        runner.target = target
        # Forward features
        runner.output = runner.model(*features)
        # Ensure `targetL` and `outputL` are always in a list format.
        targetL = [target] if not isinstance(target, (list, tuple)) else target
        outputL = [runner.output] if not isinstance(runner.output, (list, tuple)) else runner.output
        # Compute loss
        runner.loss = runner.criterion(*outputL, *targetL)
```
move this to on_iter_forward